### DISCUSSION TITLE
separate encryption to dedicated modules

### DISCUSSION THREAD

**Discussion by andreap-** [shadowsocks/shadowsocks-windows#3175] [Ideas]
New encryption algorithms (maybe not brand new, just the same algorithm with a longer key, or a variant of the same one, maybe in association with a second one, etc) will always come, and the community may argue on which ones among the oldest are to be abandoned or may still be good in particular circumstances (hardware that does have much better throughput performance with the given algorithm, etc).
Separating (or allowing to extend) the supported algorithms from the program main code will allow for a quicker adoption of new algorithms (no need of nightly/alpha builds, etc) and spare the community all of the quarreling about which are to keep and which to abandon, and at the same time keeping the code cleaner and easier to maintain

**Comment by database64128** [comment]
Maybe take a look at the code before starting a new discussion. It's already "modular" like you envisioned.

Support for the legacy Shadowsocks protocol with stream ciphers is intentionally removed. It will not be added back, period.

**Reply by andreap-** [reply]
@database64128 
thanks for the reply
I'm sorry, maybe I have not been clear enough. 
I was thinking to modular as to something you can do after the program as been compiled, like adding a .ddl in the specidic folder.  It reminds me of how it was used to add codec support to windows media player
